client/blocks/importer/wordpress/import-everything/index.tsx

```diff
@@ -32,7 +32,44 @@ interface Props {
 	targetSiteEligibleForProPlan: boolean;
 	stepNavigator?: StepNavigator;
 }
+
+interface State {
+	migrationStatus: string;
+	percent: number | null;
+}
 export class ImportEverything extends SectionMigrate {
+	componentDidUpdate( prevProps: any, prevState: State ) {
+		super.componentDidUpdate( prevProps, prevState );
+
+		if (
+			prevState.migrationStatus !== MigrationStatus.BACKING_UP &&
+			this.state.migrationStatus === MigrationStatus.BACKING_UP
+		) {
+			recordTracksEvent( 'calypso_site_importer_import_progress_backing_up' );
+		}
+
+		if (
+			prevState.migrationStatus !== MigrationStatus.RESTORING &&
+			this.state.migrationStatus === MigrationStatus.RESTORING
+		) {
+			recordTracksEvent( 'calypso_site_importer_import_progress_restoring' );
+		}
+
+		if (
+			prevState.migrationStatus !== MigrationStatus.ERROR &&
+			this.state.migrationStatus === MigrationStatus.ERROR
+		) {
+			recordTracksEvent( 'calypso_site_importer_import_failure' );
+		}
+
+		if (
+			prevState.migrationStatus !== MigrationStatus.DONE &&
+			this.state.migrationStatus === MigrationStatus.DONE
+		) {
+			recordTracksEvent( 'calypso_site_importer_import_success' );
+		}
```

Comment:
I would also suggest wrapping this code block to the semantically named function.
It is a good practice to have lifecycle methods clear and lightweight.